app/forms/prior_authority/steps/primary_quote_form.rb

```diff
@@ -13,8 +13,8 @@ class PrimaryQuoteForm < ::Steps::BaseFormObject
       validates :postcode, presence: true, uk_postcode: true
 
       def service_type_suggestion=(value)
-        if value.in?(translated_values)
-          self.service_type = service_type.value
+        if value.in?(translations.keys)
+          self.service_type = translations[value]
```

Comment:
I found this slightly confusing as expected the `value` to contact the untranslated key.... 

It looks like it is returning the translated text which it shouldn't be... ???

or am I missing something.

Reply:
@dwhenry the `value` of service_type_suggestion is the contents of the text field, which is the translated value. This is separate from the value of `service_type` which is the untranslated value, derived from the hidden HTML <select> element. The bug occurs when a user types in a value to the text field but doesn't click the autocomplete. In which case the service_type_suggestion contains a "true" translated value, but the service_type dropdown contains a false untranslated value, so the translated value needs to take precedence.